packages/backend/src/open_payments/payment/incoming/routes.ts

```diff
@@ -39,24 +38,20 @@ export function createIncomingPaymentRoutes(
   })
   const deps = { ...deps_, logger }
   return {
-    get: (ctx: AppContext) => getIncomingPayment(deps, ctx),
-    create: (ctx: AppContext) => createIncomingPayment(deps, ctx),
-    update: (ctx: AppContext) => updateIncomingPayment(deps, ctx)
+    get: (ctx: ReadContext) => getIncomingPayment(deps, ctx),
+    create: (ctx: CreateContext<CreateBody>) =>
+      createIncomingPayment(deps, ctx),
+    update: (ctx: UpdateContext<UpdateBody>) => updateIncomingPayment(deps, ctx)
   }
 }
 
 async function getIncomingPayment(
   deps: ServiceDependencies,
-  ctx: AppContext
+  ctx: ReadContext
 ): Promise<void> {
```

Comment:
It'd be nice if the route functions could assert the response body type like:
```suggestion
): Promise<ctx is ResponseContext<IncomingPaymentJSON>> {
```
but that doesn't seem possible with async functions:
- https://github.com/microsoft/TypeScript/issues/37681